test/unit/app/workers/test-adjustments-worker.js

```diff
@@ -268,10 +244,83 @@ describe(relativeFilePath, function () {
+      expect(createNewTasks.called).to.be.equal(true)
+
+      expect(stagingAdjustmentsMapper.mapCmsAdjustments.called).to.be.equal(true)
+      expect(stagingAdjustmentsMapper.mapGsAdjustments.called).to.be.equal(true)
+      expect(getAppAdjustments.called).to.be.equal(true)
+      expect(updateAdjustmentEffectiveTo.called).to.be.equal(false)
+      expect(insertAdjustment.calledWith(gsAdjustments[0])).to.be.equal(true)
+    })
+  })
+
+  it('should retrieve staging GS adjustments and update them in app when they exist but are not in the extract', function () {
+    var appAdjustments = [
+      Object.assign({}, adjustmentRow, { id: 24, points: -6, contactId: gsContactId, workloadOwnerId: existingOmId })
+    ]
+
+    stagingAdjustmentsMapper.mapCmsAdjustments.resolves([])
+    stagingAdjustmentsMapper.mapGsAdjustments.resolves([])
+    getAppAdjustments.resolves(appAdjustments)
+    updateAdjustmentEffectiveTo.resolves()
+    insertAdjustment.resolves()
+    updateAdjustmentStatusByIds.resolves(1)
+    createNewTasks.resolves()
+
+    return adjustmentsWorker.execute(task).then(function () {
+      expect(updateAdjustmentStatusByIds.called).to.be.equal(false)
+      expect(createNewTasks.called).to.be.equal(true)
+
+      expect(stagingAdjustmentsMapper.mapCmsAdjustments.called).to.be.equal(true)
+      expect(stagingAdjustmentsMapper.mapGsAdjustments.called).to.be.equal(true)
+      expect(getAppAdjustments.called).to.be.equal(true)
+      expect(updateAdjustmentEffectiveTo.calledWith(appAdjustments[0].id, updateTime)).to.be.equal(true)
+      expect(insertAdjustment.called).to.be.equal(false)
+    })
+  })
+
+  it('should correctly process a combination of adding and updating CMS and GS adjustments', function () {
+    var appAdjustments = [
+      adjustmentRow,
+      Object.assign({}, adjustmentRow, { id: 12, points: -1, workloadOwnerId: existingOmId }),
+      Object.assign({}, adjustmentRow, { id: 24, points: -6, contactId: 123, workloadOwnerId: existingOmId })
+    ]
+
+    stagingAdjustmentsMapper.mapCmsAdjustments.resolves(cmsAdjustments)
+    stagingAdjustmentsMapper.mapGsAdjustments.resolves(gsAdjustments)
+    getAppAdjustments.resolves(appAdjustments)
+    updateAdjustmentEffectiveTo.resolves()
+    insertAdjustment.resolves()
+    updateAdjustmentStatusByIds.resolves(1)
+    createNewTasks.resolves()
+
+    return adjustmentsWorker.execute(task).then(function () {
+      expect(updateAdjustmentStatusByIds.called).to.be.equal(false)
+      expect(createNewTasks.called).to.be.equal(true)
+
+      expect(stagingAdjustmentsMapper.mapCmsAdjustments.called).to.be.equal(true)
+      expect(stagingAdjustmentsMapper.mapGsAdjustments.called).to.be.equal(true)
+      expect(getAppAdjustments.called).to.be.equal(true)
+      expect(updateAdjustmentEffectiveTo.calledWith(appAdjustments[2].id, updateTime)).to.be.equal(true)
+      expect(insertAdjustment.calledWith(gsAdjustments[0])).to.be.equal(true)
+    })
+  })
 })
```

Comment:
It would also be worth testing that if any of the stubbed method calls throw an error that this is thrown by the worker, as this will be what determines that status of the task.